index.bs

```diff
@@ -2725,6 +2769,284 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
+1. If |start node parameter| is not null, set |start node| to the result of [=trying=] to
+   [=deserialize local value=] given |start node parameter|, and |session|.
+
+1. Let |wait timeout| be the value of the <code>waitTimeout</code> field of |command
+   parameters| if present, or 0 otherwise.
+
+1. Let |expiration time| be the current time plus |wait timeout| milliseconds.
+
+1. Let |result nodes| be an empty array.
+
+1. Perform the following steps until the current time is greater than |expiration time|
+   or |result nodes| is not empty.
+
+  1. Let |context nodes| be an empty array.
+
+  1. Append |start node| to |context nodes|.
+
+  1. For each |locator| in |locators|:
+
+     1. Let |type| be the value of the type property of |locator|.
+   
+     1. In the following list of conditions and associated steps, run the first set 
+        of steps for which the associated condition is true, if any:
+
+        <dl>
+
+           <dt>|type| is the string "<code>css</code>"
+           <dd>
+              1. Let |selector| be the value of the value property of |locator|.
+
+              1. Let |context nodes| be a result of [=trying=] to [=locate nodes using css=]
+                 given |context|, |context nodes|, |selector| and |session|.
+
+           <dt>|type| is the string "<code>xpath</code>"
+           <dd>
+              1. Let |selector| be the value of the value property of |locator|.
+
+              1. Let |context nodes| be a result of [=trying=] to [=locate nodes using xpath=]
+                 given |context|, |context nodes|, |selector| and |session|.
+ 
+           <dt>|type| is the string "<code>frame</code>"
+           <dd>
+              1. Let |window proxy| be the result of [=trying=] to [=locate a frame context=]
+                 given |context|, |context nodes|, |selector| and |session|.
+
+              1. Let |context| be the context property of the value property of |window proxy|.
+
+              1. Let |context nodes| be an empty array.
+
+              1. Append the node corresponding to the documentElement of |context| to
+                 |context nodes|.
+
+           <dt>|type| is the string "<code>script</code>"
+           <dd>
+              1. Let |function declaration| be the value of the function declaration
+                 property of |locator|.
+
+              1. Let |context nodes| be a result of [=trying=] to [=locate nodes using script=]
+                 given |context|, |context nodes|, |function declaration| and |session|.
+
+           <dt>|type| is the string "<code>text</code>"
+           <dd>
+              1. Let |selector| be the value of the value property of |locator|.
+
+              1. Let |match| be the value of the match property of |locator| if
+                 present, otherwise "full".
+
+              1. Let |context nodes| be a result of [=trying=] to [=locate nodes using text=]
+                 given |context|, |context nodes|, |selector|, |match| and |session|
+
```

Comment:
What happens if there is no match?

Reply:
Should be that you get an empty list returned.

Reply:
I think it may be better to indicate to the user that an `invalid selector` was used. I can imagine trying to debug a typo in a local end implementation....

Reply:
@shs96c What about the perfectly legitimate case where you're attempting to find an element by its text, and no element with that text actually exists in the document? Since we're doing the equivalent of Classic's `Find Elements` for this command, an empty list (not an exception) is the right return type.

Reply:
I agree that no match should return an empty array. Invalid selector seems strictly wrong since there are many cases (e.g. polling for an element) where a valid selector will result in zero matches.